lib/elixir/lib/kernel.ex

```diff
@@ -795,11 +798,13 @@ defmodule Kernel do
 
       iex> trunc(5.4)
       5
+
       iex> trunc(5.99)
       5
 
   """
-  @spec trunc(number) :: integer
+  @spec trunc(arg) :: arg when arg: integer
```

Comment:
The set of types returned by the anonymous function may not overlap with the set of types returned by `call_then_double/2`.